refract-spec.md

```diff
@@ -137,9 +139,490 @@ This is how it would expressed in the compact version.
+
+1. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
+1. If the content is a relative fragment identifier, it references an element found in the local document
+1. If there is no fragment identifier, the reference is to the entire document.
+1. If there are multiple elements in a document with the same ID, the first element with the matching value SHOULD be used.
+
+The `ref` element transcludes the contents of the element into the document in which it is referenced.
+
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (string) - A URL
+
+#### Examples
+
+Elements MAY be referenced in remove or local documents.
+
+##### Referencing Remote Element
+
+```json
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "#foo"
+}
+```
+
+## Namespacing
+
+Namespaces provide a way to include element definitions and constraints into another document. The following rules and constraints apply to namespaces.
+
+1. If there is a conflict, the namespaced element defined or referenced last MUST take precedence
+1. Types defined within the local namespace MUST take precedence over referenced namespaces
+1. Prefixed namespaces are only accessible through using a prefix
+1. Namespaces apply to the element in which they are introduced and all child elements
+1. Namespaces MUST not include the elements nor the data in the elements referenced, but rather include the constraints and conditions around the types defined within that namespace
+
+### Namespaces (Element)
+
+The `namespaces` element is a list of Namespace elements.
+
+#### Properties
+
+- `element` namespaces (string, fixed)
+- `content` (array[Namespace])
+
+### Namespace (enum)
+
+#### Members
+
+- (string) - A resolvable URL to a namespace
+- (object)
+    - `prefix` (string) - A prefix
+    - `href` (string) - An `href` is a resolvable URL to a namespace
+
+#### Example Using Both String and Object Namespace Elements
+
+This example shows a namespace being included by way of a string, and a prefixed namespace.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "namespaces": [
+      "http://example.com/namespace1",
+      { "prefix": "ns2", "href": "http://example.com/namespace2" }
+    ]
+  }
+}
+```
+
+### Prefix (Element)
+
+A prefix is a way to reference elements in prefixed namespaces.
+
+#### Example
+
+This example says the `foo` element is found in the prefixed namespaced prefixed as `ns2`.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "prefix": "ns2"
+  },
+  "content": "baz"
+}
+```
+
+## Extending Elements
+
+Elements by default do not include any of the attributes or content from the element in which they reference. An element MAY be extended in order to inherit from other elements.
```

Comment:
I feel this is really missing an example, or better put, the referencing element above needs an example of "The ref element transcludes the content". Maybe then this would make more sense. Can you please elaborate on it?